### PR TITLE
First lint pass

### PR DESCRIPTION
**Pull request by philtay**
Linting + Go 1.12 support

### CODE REVIEW

**Comment by beevik** [comment]
I would prefer not to drop support for go versions less than go1.12.  Many people use this package, and not all have upgraded to 1.12.

Also, what version of go lint tools are you using?  My golint didn't detect these lint changes.

File: etree.go [hunk 1]
```diff
@@ -1033,28 +1031,28 @@ func (e *Element) setIndex(index int) {
 }
 
 // writeTo serializes the element to the writer w.
-func (e *Element) writeTo(w *bufio.Writer, s *WriteSettings) {
-	w.WriteByte('<')
+func (e *Element) writeTo(w io.StringWriter, s *WriteSettings) {
+	w.WriteString("<")
```
Comment (beevik) [inline etree.go]: Why the switch to `io.StringWriter` from `bufio.Writer`?  The use of `bufio.Writer` was never exposed in the public API.  Also, there are subtle performance consequences to this change. For instance, `bufio.WriteByte` is faster than `bufio.WriteString` for single-character strings.

**Comment by philtay** [comment]
The linter is https://github.com/golangci/golangci-lint (use it in `.travis.yml` too). Keep `bufio.Writer` instead of `io.StringWriter`. An interface is more elegant but as you said is not exposed in the public API. The rest are valid changes.

**Comment by beevik** [comment]
Can you produce a new (squashed) commit without the `io.StringWriter` changes and without the removal of 1.11 support?  Also, if you like, please add yourself to the `CONTRIBUTORS` file.

**Comment by philtay** [comment]
@beevik done

**Comment by beevik** [comment]
Thank you for the submission!

**Pull request merged**